docs/dev/developer_documentation.md

```diff
@@ -60,81 +60,6 @@
 	</li>
 </ul>
 <hr>
-<br>
-<h2><a id="sec_3.1">3.1</a> How TripleA Locates its Root Directory</h2>
-<p>
-	It may be of some interest to some developers to know exactly how TripleA
-	locates where its own root directory is. This will help those who like to
-	put the start-up scripts in different directories.
-</p>
-<p>
-	The search algorithm works from the inside out and is located in the
-	<b>GameRunner.java</b> source file. It will start from the package location
-	of GameRunner.class and move up one level until it has reached the root
-	package. Once there, that will be TripleA's root directory. It then checks
-	to make sure that the root directory it found actually exists, if not then
-	return the current user's home directory and display an error message.<br>
-	<br>
-	This way the root directory doesn't have to be hard coded.
-</p>
-<table class="thickTable">
-	<tr>
-		<td><span style="font-weight: bold; font-size: small;">File Name</span>
-		</td>
-		<td><span style="font-size: small;">GameRunner.java</span></td>
-	</tr>
-		<tr>
-		<td><span style="font-weight: bold; font-size: small;">Package</span></td>
-		<td><span style="font-size: small;">games.strategy.engine.framework</span></td>
-	</tr>
-	<tr>
-		<td><span style="font-weight: bold; font-size: small;">Method Header</span></td>
-		<td><span style="font-size: small;">public static File getRootFolder()</span></td>
-	</tr>
-</table>
-<table class="thickTable">
-	<tr>
-		<td>
-			<pre><code>/**
- * Get the root folder for the application
- */
-public static File getRootFolder(){
-	//we know that the class file is in a directory one above the games root folder
-	//so navigate up from the class file, and we have root.
-	//find the url of our class
-	URL url = GameRunner.class.getResource("GameRunner.class");
-	//we want to move up 1 directory for each package
-	int moveUpCount = GameRunner.class.getName().split("\\.").length + 1;
-	String fileName = url.getFile();
-	try {
-		//deal with spaces in the file name which would be url encoded
-		fileName  = URLDecoder.decode(fileName, "UTF-8");
-	} catch (UnsupportedEncodingException e){
-		e.printStackTrace();
-	}
-	//we are in a jar file
-	if(fileName.indexOf("triplea.jar!") != -1){
-		String subString = fileName.substring("file:/".length()  - ( isWindows() ? 0 : 1)  , fileName.indexOf("triplea.jar!") -1);
-		File f = new File(subString).getParentFile();
-		if(!f.exists()){
-			throw new IllegalStateException("File not found:" + f);
-		}
-		return f;
-	} else{
-		File f = new File(fileName);
-		for(int i = 0; i &lt; moveUpCount; i++){
-			f = f.getParentFile();
-		}
-		if(!f.exists()) {
-			System.err.println("Could not find root folder, does  not exist:" + f);
```

Comment:
This line is what prompted me to remove this code block, the System.err was flagged.
A) the code being replicated here is not ideal, to maintain this we should swap it out for a github code link
B) the discussion of how we get root folder is interesting, complex, but I'm afraid out of place and it's better if we can make the code more self documenting. It's unlikely we'll keep this up to date.


In general it's perhaps about time we started folding more of this document into the code base and removing things that are well documented already or out of date.